descartes_core/src/planning_graph.cpp

```diff
@@ -29,24 +29,97 @@
 #include <iostream>
 #include <utility>
 #include <algorithm>
-#include <fstream>
 
 #include <ros/console.h>
 
 #include <boost/uuid/uuid_io.hpp> // streaming operators
 #include <boost/graph/dijkstra_shortest_paths.hpp>
 
+#include <future>
+
+
+namespace
+{
+  using JointSolutions = std::vector<std::vector<double>>;
+  using VecJointSolutions = std::vector<JointSolutions>;
+  using ConstTrajectorIterator = std::vector<descartes_core::TrajectoryPtPtr>::const_iterator;
+
+  VecJointSolutions
+  rangeCalculateJointSolutions(descartes_core::RobotModelPtr model, 
+                               ConstTrajectorIterator begin,
+                               ConstTrajectorIterator end)
+  {
+    VecJointSolutions solutions;
+    solutions.reserve(std::distance(begin, end));
+
+    while (begin != end)
+    {
+      JointSolutions joint_poses;
+      begin->get()->getJointPoses(*model, joint_poses);
+
+      solutions.push_back(std::move(joint_poses));
```

Comment:
A set of joint solutions may not always be available.  Perhaps the appropriate way to handle this would be to quit out of the async  process and print an error message as it was done in the single threaded version.